src/tests/java/htsjdk/samtools/util/SamLocusIteratorTest.java

```diff
@@ -64,6 +64,34 @@ public void testBasicIterator() {
         for (final SamLocusIterator.LocusInfo li : sli) {
             Assert.assertEquals(pos++, li.getPosition());
             Assert.assertEquals(2, li.getRecordAndPositions().size());
+            // make sure that we are not accumulating indels
+            Assert.assertEquals(li.getDeletedInRecord().size(), 0);
+            Assert.assertEquals(li.getInsertionInRecord().size(), 0);
+        }
+
+    }
+
+    @Test
+    public void testBasicIteratorWithIndels() {
+
+        final String sqHeader = "@HD\tSO:coordinate\tVN:1.0\n@SQ\tSN:chrM\tAS:HG18\tLN:100000\n";
```

Comment:
I feel bad making this comment, because this is totally not your fault.  You've made changes in a class that uses an outdated way of building SAMRecords for testing, and followed the pattern you found.  But I'd really prefer not to add lots more tests that do things this way.  Would you mind taking a look at `SAMRecordSetBuilder` and seeing if you could convert the tests to use that instead?